src/Types/BigIntType.php

```diff
@@ -22,18 +27,33 @@ public function getSQLDeclaration(array $column, AbstractPlatform $platform): st
 
     public function getBindingType(): ParameterType
     {
-        return ParameterType::STRING;
+        return ParameterType::INTEGER;
     }
 
     /**
      * @param T $value
      *
-     * @return (T is null ? null : string)
+     * @return (T is null ? null : int|string)
      *
      * @template T
      */
-    public function convertToPHPValue(mixed $value, AbstractPlatform $platform): ?string
+    public function convertToPHPValue(mixed $value, AbstractPlatform $platform): int|string|null
     {
-        return $value === null ? null : (string) $value;
+        if ($value === null) {
+            return null;
+        }
+        $options = [
+            'options' => [
+                'min_range' => PHP_INT_MIN,
+                'max_range' => PHP_INT_MAX
+            ]
+        ];
+
+        $safeInt = filter_var($value, FILTER_VALIDATE_INT, $options);
```

Comment:
When I was creating the unit tests I was getting a strange behavior.

If the value is a string and exceeds the PHP_INT_MAX then it passes the comparison.

See this PHP shell example:

```
php > var_dump(PHP_INT_MAX);
int(9223372036854775807)
php > var_dump(('9223372036854775808' <= PHP_INT_MAX));
bool(true)
```

Reply:
The problem is the `<=` operator. If you have a string with an integer lager than `PHP_INT_MAX`, an integer cast gives you `PHP_INT_MAX` in PHP.

I'm afraid the ext-filter methods are to slow to put them on a hot execution path. Let's stick to `<` and `>`. I realize that this creates an edge case for `PHP_INT_MAX` itself because we cannot tell apart `PHP_INT_MAX` from an integer overflow, but I think we can live with that.